azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -331,78 +338,211 @@ def __init__(self, xml_text):
         logger.verbose("Extension config shows status blob type as [{0}]", self.status_upload_blob_type)
 
     @staticmethod
-    def _parse_plugin(plugin):
-        ext_handler = ExtHandler()
-        ext_handler.name = getattrib(plugin, "name")
-        ext_handler.properties.version = getattrib(plugin, "version")
-        ext_handler.properties.state = getattrib(plugin, "state")
+    def _parse_plugin(ext_handler, plugin):
+
+        def _raise_config_error_if_none(attr_name, value):
+            if value is None:
+                raise ExtensionConfigError("{0} is None for ExtensionConfig, failing Extension.".format(attr_name))
+            return value
+
+        ext_handler.name = _raise_config_error_if_none("Handler Name", getattrib(plugin, "name"))
+        ext_handler.properties.version = _raise_config_error_if_none("Handler Version", getattrib(plugin, "version"))
+        ext_handler.properties.state = _raise_config_error_if_none("Handler State", getattrib(plugin, "state"))
 
         location = getattrib(plugin, "location")
         failover_location = getattrib(plugin, "failoverlocation")
         for uri in [location, failover_location]:
             version_uri = ExtHandlerVersionUri()
             version_uri.uri = uri
             ext_handler.versionUris.append(version_uri)
-        return ext_handler
 
     @staticmethod
     def _parse_plugin_settings(ext_handler, plugin_settings):
         if plugin_settings is None:
             return
 
-        name = ext_handler.name
+        handler_name = ext_handler.name
         version = ext_handler.properties.version
 
-        to_lower = lambda str_to_change: str_to_change.lower() if str_to_change is not None else None
-        ext_handler_plugin_settings = [x for x in plugin_settings if to_lower(getattrib(x, "name")) == to_lower(name)]
+        def to_lower(str_to_change): return str_to_change.lower() if str_to_change is not None else None
```

Comment:
PyCharm was complaining about this. Apparently PEP doesn't like using named-lambda functions, they are only supposed to be used as temporary functions (no-name)